### PR TITLE
HTTP dependency request tracking

### PR DESCRIPTION
**Pull request by jasongin**
I'm resubmitting this as a new PR to target the develop branch instead of the master branch. See review comments on the original PR at https://github.com/Microsoft/ApplicationInsights-node.js/pull/129. There are no changes here compared to that one, aside from rebasing on top of develop.

### CODE REVIEW

**Comment by kamilszostak** [comment]
:shipit:

File: AutoCollection/ClientRequestParser.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        let remoteDependency = new ContractsModule.Contracts.RemoteDependencyData();
+        remoteDependency.target = urlObject.hostname;
+        remoteDependency.name = dependencyName;
+        remoteDependency.commandName = this.url;
+        remoteDependency.value = this.duration;
+        remoteDependency.success = this._isSuccess();
+        remoteDependency.async = true;
+        remoteDependency.dependencyTypeName =
+            ContractsModule.Contracts.DependencyKind[ContractsModule.Contracts.DependencyKind.Http];
+        remoteDependency.dependencyKind = ContractsModule.Contracts.DependencyKind.Http;
```
Comment (brahmnes) [inline AutoCollection/ClientRequestParser.ts]: I believe a new dependency type, something called AI dependency, is being introduced, and that would be used instead of HTTP for communications between AI components.

Check with Nizar on the details.

Comment (jasongin) [inline AutoCollection/ClientRequestParser.ts]: Yes, I'll include that in the follow-up change that adds component correlation. But for now this code doesn't know when a dependency is an AI component or not.

File: AutoCollection/ServerRequests.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            return;
+        }
+
+        // store data about the request
+        var requestParser = new ServerRequestParser(request);
+
+        // response listeners
+        if (response && response.once) {
+            response.once("finish", () => {
+                AutoCollectServerRequests.endRequest(client, requestParser, response, null, properties, null);
```
Comment (brahmnes) [inline AutoCollection/ServerRequests.ts]: I just noticed that in both the Server and Client auto collectors, it's currently not parsing or generating a hashed ikey and on the HTTP headers. This will be needed for stitching different AI components in the map. Perhaps that's the next step?

Comment (jasongin) [inline AutoCollection/ServerRequests.ts]: Yes, that will be in a separate PR. I'm testing that now.
